core/modules/wsched/BlendScheduler.cc

```diff
@@ -108,27 +113,45 @@ BlendScheduler::~BlendScheduler() {
 
 
 void BlendScheduler::_sortScanSchedulers() {
-    auto greaterThan = [](SchedulerBase::Ptr const& a, SchedulerBase::Ptr const& b)->bool {
-        // Experiment of sorts, priority depends on number of Tasks in each scheduler.
-        auto aVal = a->getPriority() + a->getUserQueriesInQ();
-        auto bVal = b->getPriority() + b->getUserQueriesInQ();
-        return aVal > bVal;
+    auto lessThan = [this](SchedulerBase::Ptr const& a, SchedulerBase::Ptr const& b) -> bool {
+        // group scheduler is always first
+        if (a == _group) return true;
+        if (b == _group) return false;
```

Comment:
_group is the interactive query scheduler. Other scedulers are the ScanSchedulers (fast, medium, slow, and snail).

Reply:
My general comment (beyond a scope of this particular ticked, more like a "wish list") on the schedulers is that they are presently manually coded. It would be nice to have a generic implementation which would be externally configured. The configuration would define:
- schedulers (identified by some names like "group", "fast", "slow", etc.)
- priorities of the schedulers
- some optional parameters of the (like preemptive-ness, etc.)
- other constrains, like resources allowed to be used by each scheduler (numbers of queries, threads, memory, etc.)

Then we could experiment with different configuration options to get the most optimal performance w/o recoding Qserv each time we wanted to try something out.

Reply:
Actually, they are all ScanScheduler, except for the GroupScheduler, which has a very different behavior than the ScanScheduler. Most of the ScanScheduler parameters are already set via configuration files. They do have priorities, but priorities are used in a different way than default position (i.e. when position is not good enough). See wconfig/WorkerConfig.cc and xrdsvc::SsiService::SsiService(...).  The coding changes have to do with attempts to improve behavior in corner cases and none of them have really worked that well. The current case with _prioritizeByInFlight is that it does help with scheduler that aren't acquiring resources (memory for large tables) but it does hurt performance of the fast scheduler. I think it has some promise, but it's not useful enough as is and I'm not sure what it really needs.